packages/ui/src/components/va-carousel/VaCarousel.stories.ts

```diff
@@ -1,12 +1,203 @@
+})
+
+Default.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const carousel = document.querySelectorAll('div.va-carousel__slide')
+  const slideTwo = canvas.getByRole('button', { name: 'go slide 2' }) as HTMLElement
+  const goForward = canvas.getByRole('button', { name: 'go next slide' }) as HTMLElement
+
+  await step('Should head to slide 2', async () => {
+    userEvent.click(slideTwo)
+    await sleep()
+    expect(carousel[1].getAttribute('aria-current')).toEqual('true')
+  })
+
+  await step('Should head to slide 3 by clicking on right arrow', async () => {
+    userEvent.click(goForward)
+    await sleep()
+    expect(carousel[2].getAttribute('aria-current')).toEqual('true')
+  })
+}
+
+export const Stateful = () => ({
+  components: { VaCarousel },
+  ...carouselSetup,
+  template: `
+    [true] - should slide
+    <VaCarousel :items="items" stateful />
+    [false] - should not slide
+    <VaCarousel :items="items" />
+  `,
+})
+
+Stateful.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const carousel = document.querySelectorAll('div.va-carousel__slide')
+  const [statefulslideTwo, slideTwo] = canvas.getAllByRole('button', { name: 'go slide 2' }) as HTMLElement[]
+
+  await step('Should head to slide 2', async () => {
+    userEvent.click(statefulslideTwo)
+    await sleep()
+    expect(carousel[1].getAttribute('aria-current')).toEqual('true')
+  })
+
+  await step('Should do nothing', async () => {
+    userEvent.click(slideTwo)
+    await sleep()
+    expect(carousel[6].getAttribute('aria-current')).toEqual('false')
+  })
+}
+
+export const Vertical = () => ({
+  components: { VaCarousel, VaSwitch },
+  ...carouselSetup,
+  template: `
+    <VaCarousel :items="items" v-model="currentSlide" vertical />
+  `,
+})
+
+export const Arrows = () => ({
+  components: { VaCarousel, VaSwitch },
+  ...carouselSetup,
+  template: `
+    <va-switch
+      v-model="areArrowsVisible"
+      class="mb-6"
+      label="Arrows"
+      size="small"
+    />
+    <VaCarousel :items="items" v-model="currentSlide" :arrows="areArrowsVisible" />
+  `,
+})
+
+export const Indicators = () => ({
+  components: { VaCarousel, VaSwitch },
+  ...carouselSetup,
+  template: `
+    <div class="space-x-2 mb-6">
+      <va-switch
+        v-model="areIndicatorsVisible"
+        label="Indicators"
+        size="small"
+      />
+      <va-switch
+        v-model="triggerOnHover"
+        label="Trigger on hover"
+        size="small"
+      />
+    </div>
+    <VaCarousel :items="items" v-model="currentSlide" :indicators="areIndicatorsVisible" :indicatorTrigger="triggerOnHover ? 'hover' : 'click'" />
+  `,
+})
+
+export const Infinite = () => ({
+  components: { VaCarousel, VaSwitch },
+  ...carouselSetup,
+  template: `
+    <VaCarousel :items="items" v-model="currentSlide" infinite />
+  `,
+})
+
+Infinite.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
```

Comment:
We need to handle this similar to `stateful`.

One story with infinite (where navigate left works).
Another story without infinite (where navigate left doesn't work).